subprojects/internal-testing/src/main/groovy/org/gradle/test/preconditions/UnitTestPreconditions.groovy

```diff
@@ -523,4 +523,11 @@ class UnitTestPreconditions {
         }
     }
 
+    static final class OnRemoteTestDistributionExecutor implements TestPrecondition {
+        @Override
+        boolean isSatisfied() throws Exception {
+            return System.getenv("RUNNING_ON_REMOTE_AGENT") != null
```

Comment:
❔ Can you point to the change where you add that environment variable to the test distribution agents?